### PR TITLE
Add some natives

### PR DESCRIPTION
**Pull request by Identity-labs**

### CODE REVIEW

File: natives.json [hunk 1]
```diff
@@ -19109,20 +19109,20 @@
...
+					"name": "animDict"
 				},
 				{
-					"type": "Any",
-					"name": "p2"
+					"type": "string",
+					"name": "animName"
 				}
 			],
 			"return_type": "Any",
```
Comment (alloc8or) [inline natives.json:19128]: ``float``

File: natives.json [hunk 2]
```diff
@@ -19109,20 +19109,20 @@
...
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "Entity",
+					"name": "entity"
 				},
 				{
-					"type": "Any",
-					"name": "p1"
+					"type": "string",
```
Comment (alloc8or) [inline natives.json:19120]: ``const char*``

File: natives.json [hunk 3]
```diff
@@ -19109,20 +19109,20 @@
...
+					"name": "animDict"
 				},
 				{
-					"type": "Any",
-					"name": "p2"
+					"type": "string",
```
Comment (alloc8or) [inline natives.json:19124]: ``const char*``

File: natives.json [hunk 4]
```diff
@@ -19253,48 +19253,48 @@
...
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "Entity",
+					"name": "entity"
 				},
 				{
-					"type": "Any",
-					"name": "p1"
+					"type": "string",
```
Comment (alloc8or) [inline natives.json:19264]: ``const char*``

File: natives.json [hunk 5]
```diff
@@ -19253,48 +19253,48 @@
...
+					"name": "animDict"
 				},
 				{
-					"type": "Any",
-					"name": "p2"
+					"type": "string",
```
Comment (alloc8or) [inline natives.json:19268]: ``const char*``

File: natives.json [hunk 6]
```diff
@@ -19253,48 +19253,48 @@
...
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "Entity",
+					"name": "entity"
 				},
 				{
-					"type": "Any",
-					"name": "p1"
+					"type": "string",
```
Comment (alloc8or) [inline natives.json:19288]: ``const char*``

File: natives.json [hunk 7]
```diff
@@ -19253,48 +19253,48 @@
...
+					"name": "animDict"
 				},
 				{
-					"type": "Any",
-					"name": "p2"
+					"type": "string",
```
Comment (alloc8or) [inline natives.json:19292]: ``const char*``

File: natives.json [hunk 8]
```diff
@@ -24983,24 +24983,24 @@
...
 		"0xC349EE1E6EFA494B": {
-			"name": "_0xC349EE1E6EFA494B",
-			"comment": "",
+			"name": "_CREATE_BLOOD_POOLS_FROM_PED",
+			"comment": "Creates blood decals from ped in some interval for few seconds. Float args are guessed, not sure what they really do (but decal size really changes).",
 			"params": [
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "ped",
```
Comment (alloc8or) [inline natives.json:24990]: ``Ped``

File: natives.json [hunk 9]
```diff
@@ -27877,24 +27877,24 @@
...
 		},
 		"0x50C803A4CD5932C5": {
-			"name": "_0x50C803A4CD5932C5",
+			"name": "_SHOW_PLAYER_CORES",
 			"comment": "",
 			"params": [
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "bool",
```
Comment (alloc8or) [inline natives.json:27884]: ``BOOL``

File: natives.json [hunk 10]
```diff
@@ -27877,24 +27877,24 @@
...
 		},
 		"0xD4EE21B7CC7FD350": {
-			"name": "_0xD4EE21B7CC7FD350",
+			"name": "_SHOW_HORSE_CORES",
 			"comment": "",
 			"params": [
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "bool",
```
Comment (alloc8or) [inline natives.json:27896]: ``BOOL``

File: natives.json [hunk 11]
```diff
@@ -29118,7 +29118,7 @@
 			"build": "1207"
 		},
 		"0xF1622CE88A1946FB": {
-			"name": "_0xF1622CE88A1946FB",
+			"name": "_HIDE_PROMPT_THIS_FRAME",
```
Comment (alloc8or) [inline natives.json:29121]: More fitting name would be ``_UIPROMPT_DISABLE_PROMPTS_THIS_FRAME``.

File: natives.json [hunk 12]
```diff
@@ -67457,19 +67457,19 @@
...
 		},
 		"0xD710A5007C2AC539": {
-			"name": "_0xD710A5007C2AC539",
+			"name": "_SET_PED_COMPONENT_DISABLED",
 			"comment": "",
 			"params": [
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "ped",
```
Comment (alloc8or) [inline natives.json:67464]: ``Ped``

File: natives.json [hunk 13]
```diff
@@ -67553,7 +67553,7 @@
 			"build": "1207"
 		},
 		"0xCC8CA3E88256E58F": {
-			"name": "_0xCC8CA3E88256E58F",
+			"name": "_UPDATE_PED_VARIATION",
 			"comment": "",
```
Comment (alloc8or) [inline natives.json:67557]: Maybe add a comment here to explain what it does exactly?

File: natives.json [hunk 14]
```diff
@@ -70233,23 +70233,23 @@
...
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "Ped",
+					"name": "ped"
 				},
 				{
-					"type": "Any",
-					"name": "p1"
+					"type": "hash",
```
Comment (alloc8or) [inline natives.json:70244]: ``Hash``

File: natives.json [hunk 15]
```diff
@@ -70233,23 +70233,23 @@
...
+					"name": "component"
 				},
 				{
-					"type": "Any",
-					"name": "p2"
+					"type": "bool",
```
Comment (alloc8or) [inline natives.json:70248]: ``BOOL``

File: natives.json [hunk 16]
```diff
@@ -70233,23 +70233,23 @@
...
+					"name": "isFemale"
 				}
 			],
-			"return_type": "Any",
+			"return_type": "hash",
```
Comment (alloc8or) [inline natives.json:70252]: ``Hash``

File: natives.json [hunk 17]
```diff
@@ -89980,15 +89980,15 @@
...
 		"0x6AA3DCA2C6F5EB6D": {
-			"name": "_0x6AA3DCA2C6F5EB6D",
-			"comment": "",
+			"name": "_GET_CURRENT_ANIM_SCENE_HASH",
+			"comment": "returns false if no no anim scene is being played",
 			"params": [
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "entity",
```
Comment (alloc8or) [inline natives.json:89987]: ``Ped``

File: natives.json [hunk 18]
```diff
@@ -89980,15 +89980,15 @@
...
+					"name": "ped"
 				}
 			],
-			"return_type": "Any",
+			"return_type": "hash",
```
Comment (alloc8or) [inline natives.json:89991]: ``Hash``

File: natives.json [hunk 19]
```diff
@@ -89980,15 +89980,15 @@
 			"build": "1207"
 		},
 		"0x6AA3DCA2C6F5EB6D": {
-			"name": "_0x6AA3DCA2C6F5EB6D",
-			"comment": "",
+			"name": "_GET_CURRENT_ANIM_SCENE_HASH",
+			"comment": "returns false if no no anim scene is being played",
```
Comment (alloc8or) [inline natives.json:89984]: Please change this to ``Returns zero if no anim scene is being played.``

File: natives.json [hunk 20]
```diff
@@ -102240,107 +102240,107 @@
 	},
 	"_NAMESPACE71": {
 		"0x049D5C615BD38BAD": {
-			"name": "_0x049D5C615BD38BAD",
+			"name": "_SHOW_TOOLTIP",
 			"comment": "",
 			"params": [
 				{
-					"type": "Any",
+					"type": "struct",
```
Comment (alloc8or) [inline natives.json:102247]: For "struct" args please use ``Any*``

File: natives.json [hunk 21]
```diff
@@ -102240,107 +102240,107 @@
...
 					"name": "p0"
 				},
 				{
-					"type": "Any",
+					"type": "struct",
```
Comment (alloc8or) [inline natives.json:102251]: Same here, ``Any*``

File: natives.json [hunk 22]
```diff
@@ -102240,107 +102240,107 @@
...
 			"build": "1207"
 		},
 		"0xD05590C1AB38F068": {
-			"name": "_0xD05590C1AB38F068",
+			"name": "_SHOW_TOWN_NOTIFICATION",
 			"comment": "",
 			"params": [
 				{
-					"type": "Any",
+					"type": "struct",
```
Comment (alloc8or) [inline natives.json:102267]: ``Any*``

File: natives.json [hunk 23]
```diff
@@ -102240,107 +102240,107 @@
...
 					"name": "p0"
 				},
 				{
-					"type": "Any",
+					"type": "struct",
```
Comment (alloc8or) [inline natives.json:102271]: ``Any*``

File: natives.json [hunk 24]
```diff
@@ -102240,107 +102240,107 @@
...
 					"name": "p2"
 				}
 			],
 			"return_type": "Any",
 			"build": "1207"
 		},
 		"0xD05590C1AB38F068": {
-			"name": "_0xD05590C1AB38F068",
+			"name": "_SHOW_TOWN_NOTIFICATION",
 			"comment": "",
```
Comment (alloc8or) [inline natives.json:102264]: Short comment to explain what exactly a "town notification" is would be great.

File: natives.json [hunk 25]
```diff
@@ -102240,107 +102240,107 @@
...
 			"build": "1207"
 		},
 		"0xCEDBF17EFCC0E4A4": {
-			"name": "_0xCEDBF17EFCC0E4A4",
+			"name": "_SHOW_OBJECTIF",
 			"comment": "",
 			"params": [
 				{
-					"type": "Any",
+					"type": "struct",
```
Comment (alloc8or) [inline natives.json:102291]: ``Any*``

File: natives.json [hunk 26]
```diff
@@ -102240,107 +102240,107 @@
...
 					"name": "p0"
 				},
 				{
-					"type": "Any",
+					"type": "struct",
```
Comment (alloc8or) [inline natives.json:102295]: ``Any*``

File: natives.json [hunk 27]
```diff
@@ -102240,107 +102240,107 @@
...
+					"type": "int",
 					"name": "p3"
 				}
 			],
 			"return_type": "Any",
 			"build": "1207"
 		},
 		"0xCEDBF17EFCC0E4A4": {
-			"name": "_0xCEDBF17EFCC0E4A4",
+			"name": "_SHOW_OBJECTIF",
```
Comment (alloc8or) [inline natives.json:102287]: This is not an english word.

File: natives.json [hunk 28]
```diff
@@ -102240,107 +102240,107 @@
...
 			"build": "1207"
 		},
 		"0xB2920B9760F0F36B": {
-			"name": "_0xB2920B9760F0F36B",
+			"name": "_SHOW_SIMPLE_RIGHT_TEXT",
 			"comment": "",
 			"params": [
 				{
-					"type": "Any",
+					"type": "struct",
```
Comment (alloc8or) [inline natives.json:102311]: ``Any*``

File: natives.json [hunk 29]
```diff
@@ -102240,107 +102240,107 @@
...
 					"name": "p0"
 				},
 				{
-					"type": "Any",
+					"type": "struct",
```
Comment (alloc8or) [inline natives.json:102315]: ``Any*``

File: natives.json [hunk 30]
```diff
@@ -102240,107 +102240,107 @@
...
+					"type": "int",
 					"name": "p2"
 				}
 			],
 			"return_type": "Any",
 			"build": "1207"
 		},
 		"0xB2920B9760F0F36B": {
-			"name": "_0xB2920B9760F0F36B",
+			"name": "_SHOW_SIMPLE_RIGHT_TEXT",
```
Comment (alloc8or) [inline natives.json:102307]: What is a "simple right text"?.
Either add a comment describing it or think of a better name.

File: natives.json [hunk 31]
```diff
@@ -102240,107 +102240,107 @@
...
 			"build": "1207"
 		},
 		"0x26E87218390E6729": {
-			"name": "_0x26E87218390E6729",
+			"name": "_SHOW_ADVANCED_NOTIFICATION",
 			"comment": "",
 			"params": [
 				{
-					"type": "Any",
+					"type": "struct",
```
Comment (alloc8or) [inline natives.json:102331]: ``Any*``

File: natives.json [hunk 32]
```diff
@@ -102240,107 +102240,107 @@
...
 					"name": "p0"
 				},
 				{
-					"type": "Any",
+					"type": "struct",
```
Comment (alloc8or) [inline natives.json:102335]: ``Any*``

File: natives.json [hunk 33]
```diff
@@ -102240,107 +102240,107 @@
...
+					"type": "int",
 					"name": "p2"
 				}
 			],
 			"return_type": "Any",
 			"build": "1207"
 		},
 		"0x26E87218390E6729": {
-			"name": "_0x26E87218390E6729",
+			"name": "_SHOW_ADVANCED_NOTIFICATION",
```
Comment (alloc8or) [inline natives.json:102327]: Again, what makes this one ''advanced"? Add a comment please.

File: natives.json [hunk 34]
```diff
@@ -102376,23 +102376,23 @@
 			"build": "1207"
 		},
 		"0xC927890AA64E9661": {
-			"name": "_0xC927890AA64E9661",
+			"name": "_SHOW_NOTIFICATION",
 			"comment": "",
 			"params": [
 				{
-					"type": "Any",
+					"type": "struct",
```
Comment (alloc8or) [inline natives.json:102383]: ``Any*``

File: natives.json [hunk 35]
```diff
@@ -102376,23 +102376,23 @@
...
 					"name": "p0"
 				},
 				{
-					"type": "Any",
+					"type": "struct",
```
Comment (alloc8or) [inline natives.json:102387]: ``Any*``

File: natives.json [hunk 36]
```diff
@@ -115029,15 +115029,15 @@
 			"build": "1207"
 		},
 		"0xD3750CCC00635FC2": {
-			"name": "_0xD3750CCC00635FC2",
+			"name": "_GET_WEAPON_MAGAZINE_CAPACITY",
```
Comment (alloc8or) [inline natives.json:115032]: Please change this to ``_GET_WEAPON_CLIP_SIZE``

File: natives.json [hunk 37]
```diff
@@ -115029,15 +115029,15 @@
...
 			"comment": "",
 			"params": [
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "weaponHash",
```
Comment (alloc8or) [inline natives.json:115036]: ``Hash``

File: natives.json [hunk 38]
```diff
@@ -115441,127 +115441,127 @@
...
 		},
 		"0x904103D5D2333977": {
-			"name": "_0x904103D5D2333977",
+			"name": "_GET_WEAPON_RUST_LEVEL",
 			"comment": "",
 			"params": [
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "entity",
```
Comment (alloc8or) [inline natives.json:115448]: ``Entity``

File: natives.json [hunk 39]
```diff
@@ -115441,127 +115441,127 @@
...
 		},
 		"0xE22060121602493B": {
-			"name": "_0xE22060121602493B",
+			"name": "_SET_WEAPON_RUST_LEVEL",
 			"comment": "",
 			"params": [
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "entity",
```
Comment (alloc8or) [inline natives.json:115460]: ``Entity``

File: natives.json [hunk 40]
```diff
@@ -115441,127 +115441,127 @@
...
 				},
 				{
-					"type": "Any",
-					"name": "p1"
+					"type": "float",
+					"name": "level"
 				},
 				{
-					"type": "Any",
+					"type": "bool",
```
Comment (alloc8or) [inline natives.json:115468]: ``BOOL``

File: natives.json [hunk 41]
```diff
@@ -115441,127 +115441,127 @@
...
 		},
 		"0x810E8AE9AFEA7E54": {
-			"name": "_0x810E8AE9AFEA7E54",
+			"name": "_GET_WEAPON_DIRT_LEVEL",
 			"comment": "",
 			"params": [
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "entity",
```
Comment (alloc8or) [inline natives.json:115480]: ``Entity``

File: natives.json [hunk 42]
```diff
@@ -115441,127 +115441,127 @@
...
 		},
 		"0x812CE61DEBCAB948": {
-			"name": "_0x812CE61DEBCAB948",
+			"name": "_SET_WEAPON_DIRT_LEVEL",
 			"comment": "",
 			"params": [
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "entity",
```
Comment (alloc8or) [inline natives.json:115492]: ``Entity``

File: natives.json [hunk 43]
```diff
@@ -115441,127 +115441,127 @@
...
 				},
 				{
-					"type": "Any",
-					"name": "p1"
+					"type": "float",
+					"name": "level"
 				},
 				{
-					"type": "Any",
+					"type": "bool",
```
Comment (alloc8or) [inline natives.json:115500]: ``BOOL``

File: natives.json [hunk 44]
```diff
@@ -115441,127 +115441,127 @@
...
 		},
 		"0x4BF66F8878F67663": {
-			"name": "_0x4BF66F8878F67663",
+			"name": "_GET_WEAPON_MUD_LEVEL",
 			"comment": "",
 			"params": [
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "entity",
```
Comment (alloc8or) [inline natives.json:115512]: ``Entity``

File: natives.json [hunk 45]
```diff
@@ -115441,127 +115441,127 @@
...
 		},
 		"0xA9EF4AD10BDDDB57": {
-			"name": "_0xA9EF4AD10BDDDB57",
+			"name": "_SET_WEAPON_MUD_LEVEL",
 			"comment": "",
 			"params": [
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "entity",
```
Comment (alloc8or) [inline natives.json:115524]: ``Entity``

File: natives.json [hunk 46]
```diff
@@ -115441,127 +115441,127 @@
...
 				},
 				{
-					"type": "Any",
-					"name": "p1"
+					"type": "float",
+					"name": "level"
 				},
 				{
-					"type": "Any",
+					"type": "bool",
```
Comment (alloc8or) [inline natives.json:115532]: ``BOOL``

File: natives.json [hunk 47]
```diff
@@ -115441,127 +115441,127 @@
...
 		"0xD4071EFC83794B2F": {
-			"name": "_0xD4071EFC83794B2F",
-			"comment": "",
+			"name": "_SET_WEAPON_LEVEL_THRESHOLD",
+			"comment": "every other level will have the max value of (brokeLevel - threshold)",
 			"params": [
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "entity",
```
Comment (alloc8or) [inline natives.json:115544]: ``Entity``

File: natives.json [hunk 48]
```diff
@@ -115441,127 +115441,127 @@
...
 		"0x0D78E1097F89E637": {
-			"name": "_0x0D78E1097F89E637",
-			"comment": "",
+			"name": "_GET_WEAPON_BROKEN_LEVEL",
+			"comment": "0.0: GOOD CONDITION 1.0: POOR CONDITION",
 			"params": [
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "entity",
```
Comment (alloc8or) [inline natives.json:115560]: ``Entity``

File: natives.json [hunk 49]
```diff
@@ -115577,16 +115577,16 @@
...
 		},
 		"0xA7A57E89E965D839": {
-			"name": "_0xA7A57E89E965D839",
+			"name": "_SET_WEAPON_BROKEN_LEVEL",
 			"comment": "",
 			"params": [
 				{
-					"type": "Any",
-					"name": "p0"
+					"type": "entity",
```
Comment (alloc8or) [inline natives.json:115584]: ``Entity``

File: natives.json [hunk 50]
```diff
@@ -115441,127 +115441,127 @@
...
+					"name": "threshold"
 				}
 			],
 			"return_type": "void",
 			"build": "1207"
 		},
 		"0x0D78E1097F89E637": {
-			"name": "_0x0D78E1097F89E637",
-			"comment": "",
+			"name": "_GET_WEAPON_BROKEN_LEVEL",
```
Comment (alloc8or) [inline natives.json:115556]: ``_GET_WEAPON_CONDITION`` probably more fitting

File: natives.json [hunk 51]
```diff
@@ -115577,16 +115577,16 @@
 			"build": "1207"
 		},
 		"0xA7A57E89E965D839": {
-			"name": "_0xA7A57E89E965D839",
+			"name": "_SET_WEAPON_BROKEN_LEVEL",
```
Comment (alloc8or) [inline natives.json:115580]: ``_SET_WEAPON_CONDITION``

**Comment by Identity-labs** [comment]
All comments has been handled, I have add exemple for natives using struct

File: natives.json [hunk 52]
```diff
@@ -102240,111 +102240,111 @@
...
 					"name": "p1"
 				},
 				{
-					"type": "Any",
+					"type": "int",
 					"name": "p2"
 				}
 			],
-			"return_type": "Any",
+			"return_type": "void",
```
Comment (alloc8or) [inline natives.json:102259]: Please _do not_ ever change something to ``void``. Please change this back to ``Any``.

File: natives.json [hunk 53]
```diff
@@ -102240,111 +102240,111 @@
...
 					"name": "p2"
 				},
 				{
-					"type": "Any",
+					"type": "int",
 					"name": "p3"
 				}
 			],
-			"return_type": "Any",
+			"return_type": "void",
```
Comment (alloc8or) [inline natives.json:102283]: Please change this back to ``Any``.

File: natives.json [hunk 54]
```diff
@@ -102240,111 +102240,111 @@
...
 					"name": "p1"
 				},
 				{
-					"type": "Any",
+					"type": "int",
 					"name": "p2"
 				}
 			],
-			"return_type": "Any",
+			"return_type": "void",
```
Comment (alloc8or) [inline natives.json:102303]: ``Any``

File: natives.json [hunk 55]
```diff
@@ -102240,111 +102240,111 @@
...
```
Comment (alloc8or) [inline natives.json:102323]: ``Any``

File: natives.json [hunk 56]
```diff
@@ -102240,111 +102240,111 @@
...
 					"name": "p2"
 				},
 				{
-					"type": "Any",
+					"type": "int",
 					"name": "p3"
 				}
 			],
-			"return_type": "Any",
+			"return_type": "void",
```
Comment (alloc8or) [inline natives.json:102347]: ``Any``

**Comment by Identity-labs** [comment]
> Seems good so far except the instances where you changed the return type to `void` (never do that!).

May be PUT readme with 
- code style
- every type to use
- best practice

**Pull request merged**